cassandra-mesos-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/SeedManager.java

```diff
@@ -0,0 +1,110 @@
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.CassandraFrameworkConfiguration;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.ExternalDc;
+
+public class SeedManager {
+    public static final long FETCH_PERIOD = TimeUnit.MINUTES.toMillis(1);
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeedManager.class);
+
+    @NotNull
+    private final PersistedCassandraFrameworkConfiguration configuration;
+
+    public SeedManager(@NotNull PersistedCassandraFrameworkConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @NotNull
+    public List<String> getSeeds() {
+        final List<String> seeds = newArrayList();
+
+        for (final ExternalDc dc : configuration.get().getExternalDcsList()) {
+            seeds.addAll(dc.getSeedsList());
+        }
+
+        return seeds;
+    }
+
+    public void syncSeeds(long now) {
+        for (ExternalDc dc : configuration.get().getExternalDcsList()) {
+            fetchSeedsIfRequired(dc, now);
+        }
+    }
+
+    private void fetchSeedsIfRequired(ExternalDc dc, long now) {
+        long lastTime = dc.getSeedFetchTime();
+        if (now - lastTime < FETCH_PERIOD) return;
+
+        String url = dc.getUrl();
+        if (url.endsWith("/")) url = url.substring(0, url.length() - 1);
+        url += "/node/seed/all";
+
+        JsonNode node = fetchJson(url);
+        if (node == null) return;
+
+        JsonNode seedsArr = node.get("seeds");
+        List<String> seeds = newArrayList();
+        for (JsonNode ip : seedsArr) {
+            seeds.add(ip.asText());
+        }
+
+        LOGGER.info("Got seeds for dc \"" + dc.getName() + "\": " + Joiner.on(", ").join(seeds));
+        updateDcSeeds(dc, seeds, now);
+    }
+
+    protected JsonNode fetchJson(String url) {
+        HttpURLConnection c = null;
+        try {
+            c = (HttpURLConnection)new URL(url).openConnection();
+            ObjectMapper mapper = new ObjectMapper();
+            return mapper.readTree(c.getInputStream());
+        } catch (IOException e) {
+            LOGGER.warn("", e);
+        } finally {
+            if (c != null) c.disconnect();
+        }
+
+        return null;
+    }
+
+    private int externalDxIdx(final String name) {
+        final List<ExternalDc> dcs = configuration.get().getExternalDcsList();
+
+        for (int i = 0; i < dcs.size(); i++) {
+            ExternalDc dc = dcs.get(i);
+            if (dc.getName().equals(name)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void updateDcSeeds(final ExternalDc dc, final List<String> seeds, final long time) {
+        int idx = externalDxIdx(dc.getName());
+        if (idx == -1) throw new IllegalArgumentException("Unknown dc " + dc.getName());
```

Comment:
Add curly braces to `if` statement and move statement to it's own line.

Also, is there some other way we can deal with this other than throwing an exception? This code is confusing to me, can you explain what it's doing?